### ISSUE TITLE
[v0.10.2] refresh can't detect sql-server and sql database were removed

### ISSUE THREAD

**Issue by 810123** [terraform-providers/terraform-provider-azurerm#255]
### Terraform Version
0.10.2

### Affected Resource(s)
Please list the resources as a list, for example:
azurerm_sql_server.server
azurerm_sql_database.db

### Terraform Configuration Files
```hcl
resource "azurerm_resource_group" "rg" {
  name     = "${var.resource_group}"
  location = "${var.location}"
}

resource "azurerm_sql_database" "db" {
  name                             = "mysqldatabase"
  resource_group_name              = "${azurerm_resource_group.rg.name}"
  location                         = "${var.location}"
  edition                          = "Basic"
  collation                        = "SQL_Latin1_General_CP1_CI_AS"
  create_mode                      = "Default"
  requested_service_objective_name = "Basic"
  server_name                      = "${azurerm_sql_server.server.name}"
}

resource "azurerm_sql_server" "server" {
  name                         = "${var.resource_group}-sqlsvr"
  resource_group_name          = "${azurerm_resource_group.rg.name}"
  location                     = "${var.location}"
  version                      = "12.0"
  administrator_login          = "${var.sql_admin}"
  administrator_login_password = "${var.sql_password}"
}

resource "azurerm_sql_firewall_rule" "fw" {
  name                = "firewallrules"
  resource_group_name = "${azurerm_resource_group.rg.name}"
  server_name         = "${azurerm_sql_server.server.name}"
  start_ip_address    = "0.0.0.0"
  end_ip_address      = "0.0.0.0"
}

```

### Debug Output
Manually delete sqldatabase and sqlserver from azure portal, then run terraform refresh command. It said the sources were not found.
```
azurerm_resource_group.production: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-a20de02f24d5/resourceGroups/production)
azurerm_resource_group.rg: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-a20de02f24d5/resourceGroups/deploy2)
azurerm_sql_server.server: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-...s/Microsoft.Sql/servers/deploy2-sqlsvr)
azurerm_virtual_network.network: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-...work/virtualNetworks/productionNetwork)
Error refreshing state: 1 error(s) occurred:

* azurerm_sql_server.server: 1 error(s) occurred:

* azurerm_sql_server.server: azurerm_sql_server.server: Error reading SQL Server: ResourceNotFound (404) - The Resource 'Microsoft.Sql/servers/deploy2-sqlsvr' under resource group 'deploy2' was not found.
```

```
azurerm_resource_group.production: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-a20de02f24d5/resourceGroups/production)
azurerm_resource_group.rg: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-a20de02f24d5/resourceGroups/deploy2)
azurerm_virtual_network.network: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-...work/virtualNetworks/productionNetwork)
azurerm_sql_server.server: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-...s/Microsoft.Sql/servers/deploy2-sqlsvr)
azurerm_sql_firewall_rule.fw: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-...oy2-sqlsvr/firewallRules/firewallrules)
azurerm_sql_database.db: Refreshing state... (ID: /subscriptions/ac4f6769-2698-4179-b24d-...deploy2-sqlsvr/databases/mysqldatabase)
Error refreshing state: 1 error(s) occurred:

* azurerm_sql_database.db: 1 error(s) occurred:

* azurerm_sql_database.db: azurerm_sql_database.db: Error reading SQL Database: ResourceNotFound (404) - The Resource 'Microsoft.Sql/servers/deploy2-sqlsvr/databases/mysqldatabase' under resource group 'deploy2' was not found.
```

### Expected Behavior
The removed service or resource should be detected.

**Comment by dominik-lekse** [comment]
I experienced the same issue. This will be fixed in #189, #190, and #191 .

**Comment by tombuildsstuff** [comment]
This will be fixed in #289
